### PR TITLE
Add in initial LETKFOI snow DA cycling capability 

### PR DESCRIPTION
**Pull request by jiaruidong2017**
**Add Jobs and Scripts for JEDI-Based Land DA**
To support global land initialization for future GFS implementations, we need to add to global-workflow jobs and scripts similar to what have already been created for JEDI-based aerosol/atmospheric DA, but for a separate, land analysis.

- jobs/JGDAS_GLOBAL_LAND_ANALYSIS_FINALIZE
- jobs/JGDAS_GLOBAL_LAND_ANALYSIS_INITIALIZE
- jobs/JGDAS_GLOBAL_LAND_ANALYSIS_RUN
- scripts/exgdas_global_land_analysis_finalize.py
- scripts/exgdas_global_land_analysis_initialize.py
- scripts/exgdas_global_land_analysis_run.sh

To support global land initialization for future GFS implementations, we need to add to global-workflow experiment setup and rocoto XML generation scripts support for including land DA related tasks. This PR adds in stubs for land DA tasks (jobs/rocoto shell scripts), the ability for setup_expt and setup_xml to include three land DA related tasks.

ADD the following 3 new tasks/jobs: 
- jobs/rocoto/landanlinit
- jobs/rocoto/landanlrun
- jobs/rocoto/landanlfinal

The prep job should:
- convert netCDF GTS snow depth obs to IODA format using python/compiled code
- stage backgrounds, fix files, and all other things necessary to run an land analysis.

Run job:
- create YAML file from template(s)
- Run fv3jedi_letkf.x

Post job:
- archive diagnostic files
- move/copy increments/analysis files and make changes as needed to stage for model initialization

Acceptance Criteria (Definition of Done)
- A 'dummy' cycling system is able to run with Land DA tasks included.

**Checklist**

- [x]  My code follows the style guidelines of this project
- [x]  I have performed a self-review of my own code
- [x]  I have commented my code, particularly in hard-to-understand areas
- [x]  My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x]  My changes generate no new warnings
- [x]  New and existing tests pass with my changes
- [x]  Any dependent changes have been merged and published

### CODE REVIEW

File: parm/parm_gdas/land_jedi_fix.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+mkdir:
+- !ENV ${DATA}/fv3jedi
```
Comment (RussTreadon-NOAA) [inline parm/parm_gdas/land_jedi_fix.yaml]: g-w is moving away from this structure.   Look at `parm/parm_gdas/atm_jedi_fix.yaml` for new structure.

File: scripts/exglobal_aero_analysis_finalize.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env python3
-# exgdas_global_aero_analysis_finalize.py
+# exglobal_aero_analysis_finalize.py
```
Comment (RussTreadon-NOAA) [inline scripts/exglobal_aero_analysis_finalize.py:2]: Good catch.  The same typo is in all three `exglobal_atm_analysis*` and all three `exglobal_atmens_analysis*` python scripts.

`exglobal_aero_analysis_run.py` has the same typo.  The comment
```
#!/usr/bin/env python3
# exgdas_global_aero_analysis_run.py
# This script creates an AerosolAnalysis object
```
should read
```
#!/usr/bin/env python3
# exglobal_aero_analysis_run.py
# This script creates an AerosolAnalysis object
```

Comment (jiaruidong2017) [reply scripts/exglobal_aero_analysis_finalize.py:2]: @RussTreadon-NOAA Do you want me to correct the typo in this PR? Thanks.

File: scripts/exglobal_atmos_sfcanl.sh [hunk 3]
```diff
@@ -194,7 +194,8 @@ fi
 
 # Update surface restarts at middle of window
 for n in $(seq 1 ${ntiles}); do
-    ${NCP} ${COMIN_GES}/RESTART/${PDY}.${cyc}0000.sfc_data.tile${n}.nc ${COMOUT}/RESTART/${PDY}.${cyc}0000.sfcanl_data.tile${n}.nc
+#DONG    ${NCP} ${COMIN_GES}/RESTART/${PDY}.${cyc}0000.sfc_data.tile${n}.nc ${COMOUT}/RESTART/${PDY}.${cyc}0000.sfcanl_data.tile${n}.nc
```
Comment (RussTreadon-NOAA) [inline scripts/exglobal_atmos_sfcanl.sh]: Why do we retain this commented out line?

File: scripts/exglobal_land_analysis_run.sh [hunk 4]
```diff
@@ -0,0 +1,174 @@
+#!/bin/bash
```
Comment (RussTreadon-NOAA) [inline scripts/exglobal_land_analysis_run.sh]: I think the g-w preference is to use python for UFS-DA ex-scripts.

File: scripts/exglobal_atmos_sfcanl.sh (hunk 3 above)
Comment (jiaruidong2017) [reply scripts/exglobal_atmos_sfcanl.sh]: @RussTreadon-NOAA This is just for my notes in this draft PR, and I will remove it. Thanks.

**Review by RussTreadon-NOAA — COMMENTED** [review]
I did not closely examine `ush/python/pygfs/task/land_analysis.py`.   I also did not check to see if the `workflow` updates generate valid xml with the correct job configurations and dependencies.

This is a draft so I expect additional changes will be made before it is ready for review.

File: scripts/exglobal_aero_analysis_finalize.py (hunk 2 above)
Comment (RussTreadon-NOAA) [reply scripts/exglobal_aero_analysis_finalize.py:2]: @jiaruidong2017 , if you want to correct typos in `atm` and `atmens` python scripts, please do so.   Thanks for asking.

Comment (jiaruidong2017) [reply scripts/exglobal_aero_analysis_finalize.py:2]: @RussTreadon-NOAA Done the corrections of these typos. Thanks.

Comment (RussTreadon-NOAA) [reply scripts/exglobal_aero_analysis_finalize.py:2]: Thanks @jiaruidong2017

File: scripts/exglobal_atmos_sfcanl.sh (hunk 3 above)
Comment (RussTreadon-NOAA) [reply scripts/exglobal_atmos_sfcanl.sh]: Good to know.  Thanks @jiaruidong2017

File: scripts/exglobal_land_analysis_run.sh (hunk 4 above)
Comment (jiaruidong2017) [reply scripts/exglobal_land_analysis_run.sh]: @RussTreadon-NOAA There are many processes/runs in this run script `scripts/exglobal_land_analysis_run.sh` for the snow DA, such as: 1) create the ensembles; 2) process the IMS data; 3) conduct JEDI runs; and 4) apply JEDI increments. We have discussed and agreed to use the shell scripts here for this snow DA implementation initially due to the limited time.

Comment (RussTreadon-NOAA) [reply scripts/exglobal_land_analysis_run.sh]: If the g-w team approves of this approach, I'm fine with their decision.

File: parm/parm_gdas/land_jedi_fix.yaml (hunk 1 above)
Comment (jiaruidong2017) [reply parm/parm_gdas/land_jedi_fix.yaml]: Updated to the new code structure.

**Comment by jiaruidong2017** [comment]
This PR is closed and moved to the new opened PR #1541

**Pull request closed without merging**